src/citrine/informatics/predictors.py

```diff
@@ -159,21 +162,43 @@ class GraphPredictor(Serializable['GraphPredictor'], Predictor):
     def __init__(self,
                  name: str,
                  description: str,
-                 predictors: List[UUID],
+                 predictors: List[Union[UUID, Predictor]],
                  session: Optional[Session] = None,
                  report: Optional[Report] = None,
                  active: bool = True):
         self.name: str = name
         self.description: str = description
-        self.predictors: List[UUID] = predictors
+        self.predictors: List[Union[UUID, Predictor]] = predictors
         self.session: Optional[Session] = session
         self.report: Optional[Report] = report
         self.active: bool = active
 
     def _post_dump(self, data: dict) -> dict:
         data['display_name'] = data['config']['name']
+        for i, predictor in enumerate(data['config']['predictors']):
+            if isinstance(predictor, dict):
+                # embedded predictors are not modules, so only serialize their config
+                data['config']['predictors'][i] = predictor['config']
+        return data
+
+    @classmethod
+    def _pre_build(cls, data: dict) -> dict:
+        for i, predictor in enumerate(data['config']['predictors']):
+            if isinstance(predictor, dict):
+                data['config']['predictors'][i] = \
+                    GraphPredictor.stuff_predictor_into_envelope(predictor)
         return data
 
+    @staticmethod
+    def stuff_predictor_into_envelope(predictor: dict) -> dict:
+        """Insert a serialized embedded predictor into a module envelope, to facilitate deser."""
+        return dict(
+            module_type='PREDICTOR',
+            config=predictor,
+            active=False,
+            schema_id='43c61ad4-7e33-45d0-a3de-504acb4e0737'  # TODO: what should this be?
```

Comment:
I don't understand the function of schema_id, so I'm not sure what to do about this. Right now it's required for deserializing all predictors. Options:
1. Make it optional for predictors except the graph predictor (like we do in this PR for `status`)
2. Insert an arbitrary value like is done here (this is the schema_id for graph predictors, so probably wouldn't want to use that)
3. Check to see what type of predictor this is and insert the relevant schema_id